src/environmentPath.ts

```diff
@@ -90,6 +98,19 @@ export class Environment {
       }
     }
 
+    if (!this.PATH && this.isPortable) {
+      if (process.platform === "darwin") {
+        this.PATH = process.env.HOME + "/Library/Application Support";
+        this.OsType = OsType.Mac;
+      } else if (process.platform === "linux") {
+        this.PATH = process.env.VSCODE_PORTABLE;
+        this.OsType = OsType.Linux;
+      } else {
+        this.PATH = process.env.VSCODE_PORTABLE;
+        this.OsType = OsType.Linux;

```

Comment:
should this be OsType.Windows ? (Just browsing the code)

Reply:
Very nice, Yes!
I will fix it.

Feel free to correct / improve where required.
As this update is huge and although i have tested well -  i still expect issues 😄